pytential/symbolic/mappers.py

```diff
@@ -160,13 +192,22 @@ def map_variable(self, expr):
         return expr
 
     def map_subscript(self, expr):
-        return self.rec(expr.aggregate)[self.rec(expr.index)]
+        aggregate = self.rec(expr.aggregate)
+        index = self.rec(expr.index)
+        if aggregate is expr.aggregate and index is expr.index:
+            return expr
+
+        return aggregate[index]
 
     map_q_weight = map_variable
     map_ones = map_variable
 
     def map_node_sum(self, expr):
-        return componentwise(type(expr), self.rec(expr.operand))
```

Comment:
This `componentwise` is not needed because `NodeSum` already does it in its `__new__`